src/entity/BookmarkList.ts

```diff
@@ -9,6 +20,26 @@ export class BookmarkList {
   @Index()
   userId: string;
 
+  @ManyToOne('User', {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  user: Promise<User>;
+
   @Column({ type: 'text' })
   name: string;
+
+  @Column({ type: 'text', nullable: true })
+  icon: string;
+
+  @CreateDateColumn()
+  createdAt: Date;
+
+  @UpdateDateColumn()
+  updatedAt: Date;
+
+  @OneToMany(() => Bookmark, (bookmark) => bookmark.listId, {
+    lazy: true,
+  })
+  bookmarks: Promise<Bookmark[]>;
```

Comment:
```suggestion
  @OneToMany('Bookmark', (bookmark: Bookmark) => bookmark.listId, {
    lazy: true,
  })
  bookmarks: Promise<Bookmark[]>;
```

Reply:
Also if you can adjust existing relations here and in `Bookmark` entity to use this kind of pattern. It helps with circular dependencies.

Reply:
Updated ✔️